src/HttpMethodsClient.php

```diff
@@ -24,7 +24,7 @@
  * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
  * @author David Buchmann <mail@davidbu.ch>
  */
-class HttpMethodsClient implements HttpClient
+final class HttpMethodsClient implements HttpClient
```

Comment:
Same problem as the BatchClient here, not possible to mock public methods of this class, we should use an annotation here instead.